pkg/storage/access_token_memory.go

```diff
@@ -1,25 +1,42 @@
 package storage
 
-import "github.com/cyberark/conjur-authn-k8s-client/pkg/storage/config"
+import (
+	"fmt"
+	"github.com/cyberark/conjur-authn-k8s-client/pkg/storage/config"
+)
 
 type AccessTokenMemory struct {
-	Data	[]byte
+	Data []byte
 }
 
 func NewAccessTokenMemory(config config.Config) (token *AccessTokenMemory, err error) {
-	return &AccessTokenMemory {
-
+	return &AccessTokenMemory{
+		Data: nil,
 	}, nil
 }
 
 func (token AccessTokenMemory) Read() (Data []byte, err error) {
-	return nil,nil
+	if token.Data == nil {
+		return nil, fmt.Errorf("error reading access token, reason: data is empty")
+	}
+
+	return token.Data, nil
 }
 
-func (token *AccessTokenMemory) Write() (err error) {
+func (token *AccessTokenMemory) Write(Data []byte) (err error) {
+	if Data == nil {
+		return fmt.Errorf("error writing access token, reason: data is empty")
+	}
+
+	token.Data = Data
 	return nil
 }
 
 func (token *AccessTokenMemory) Delete() (err error) {
+	// Clear Data
+	empty := make([]byte, len(token.Data))
+	copy(token.Data, empty)
```

Comment:
what happens to `empty`? is this to clear the token from the memory or just from the `AccessToken` object? if it's the second one, why do we do this?

Reply:
the first one - to clear the data from memory, do you think i should change here something ?

Reply:
how does it clear the data from memory? is `empty` handled by the garbage collector at the end of the method? that's the idea?

Reply:
empty - is an empty slice, and i copy this empty slice to Data